mcs/class/System/Mono.AppleTls/AppleTlsContext.cs

```diff
@@ -73,8 +74,7 @@ class AppleTlsContext : MobileTlsContext
 			: base (parent, serverMode, targetHost, enabledProtocols,
 				serverCertificate, clientCertificates, askForClientCert)
 		{
-			handle = GCHandle.Alloc (this);
-			connectionId = GCHandle.ToIntPtr (handle);
+			handle = GCHandle.Alloc (this, GCHandleType.Weak);
```

Comment:
Does this change mean that the `this` object is now eligible for collection while we're in native code performing reads/writes? If it *is* eligible then it sounds like we've broken our guarantee. If it's *not* eligible for GC then the GC handle seems unnecessary.